src/invariant/LedgerEntryIsValid.cpp

```diff
@@ -108,6 +108,8 @@ LedgerEntryIsValid::checkIsValid(LedgerEntry const& le,
         return checkIsValid(le.data.data(), version);
     case CLAIMABLE_BALANCE:
         return checkIsValid(le, previous, version);
+    case LIQUIDITY_POOL:
+        return checkIsValid(le, previous, version);
```

Comment:
This is a bug. It should be `le.data.liquidityPool()`. It compiles because we use a `checkIsValid` that accepts a `LedgerEntry` for `CLAIMABLE_BALANCE`. We should change that signature to take a `ClaimableBalanceEntry`, and just do the sponsorship validation in this switch statement.